### DISCUSSION TITLE
Missing Metadata in RetrieveUserProxyAgent Retrieval Context

### DISCUSSION THREAD

**Discussion by Naitik-k** [microsoft/autogen#3956] [Q&A]
_Description_:
The **RetrieveUserProxyAgent** class does not pass **metadata** (such as document source, title, etc.) into the context when retrieving data. This metadata can be crucial for understanding the context of the retrieved content.

_Question_:
Is there any existing mechanism to include this metadata in the context during retrieval, or could this feature be added in future versions?

**Comment by thinkall** [comment]
Thank you @Naitik-k . Source has been added in #2289 , other metadata could be added as well. The reason we didn't add all the possible metadata into the context was that it wouldn't always improve the results. Nevertheless, it's a good idea to add more support to pass metadata into the context for different scenarios. Would you like to raise a PR for it?

**Reply by Naitik-k** [reply]
Yes @thinkall 
It would be helpful in our use case if metadata(source, title etc.) can be passed in context

**Reply by Naitik-k** [reply]
@thinkall 
I'm asking this because the RetrieveUserProxyAgent class works fine for a single document, but when multiple documents with similar content are used, the LLM becomes confused. To reduce this confusion, metadata (including the source and title) should be passed as context to the LLM.
